### PR TITLE
Decompiled 2 functions in iMath3.cpp as well as partaily iSphereIsectRay

### PR DESCRIPTION
**Pull request by sonich2401**
Decompiled 2 Functions as well as partailly iSphereIsectRay. Also in progress to decompile Vec3Copy

### CODE REVIEW

File: src/Core/p2/iMath3.cpp [hunk 1]
```diff
@@ -1,16 +1,115 @@
 #include "iMath3.h"
-
+#include "../x/xVec3Inlines.h"
 #include <types.h>
 
 // func_800C4F18
-#pragma GLOBAL_ASM("asm/Core/p2/iMath3.s", "iMath3Init__Fv")
+void iMath3Init(){
 
+}
```
Comment (mattbruv) [inline src/Core/p2/iMath3.cpp:8]: Make sure you format every file you commit with clang-format with our ruleset

File: .gitignore [hunk 2]
```diff
@@ -19,3 +19,15 @@ tools/elf2dol
...
+ldscript.lcf
+asmdiff.sh
+.clang-format
+SpongeBob SquarePants - Battle for Bikini Bottom.7z
+ldscript.lcf
+asmdiff.sh
+.clang-format
+ldscript.lcf
+asmdiff.sh
+.clang-format
```
Comment (mattbruv) [inline .gitignore]: I don't know if a program generated this stuff or if you handwrote it, but none of this should be here. A lot of those things are scripts that we do want tracked by git if they are changed.

File: src/Core/p2/iMath3.cpp [hunk 3]
```diff
@@ -1,16 +1,115 @@
...
 // func_800C4F1C
-#pragma GLOBAL_ASM("asm/Core/p2/iMath3.s", "iSphereIsectVec__FPC7xSpherePC5xVec3P6xIsect")
-
+void iSphereIsectVec(const xSphere *param_1,const xVec3 *param_2,xIsect *param_3)
+{
+  float fVar1;
+  xVec3Sub(&param_3->norm,param_2,&param_1->center);
+  fVar1 = xVec3Length(&param_3->norm);
+  param_3->dist = (float)fVar1;
```
Comment (mattbruv) [inline src/Core/p2/iMath3.cpp:15]: use our custom typedefs: i.e. `float32` instead of `float`

File: src/Core/p2/iMath3.cpp [hunk 4]
```diff
@@ -1,16 +1,115 @@
...
+        }
+      }
+    }
+  }
+  __psq_l0(auStack8,in_GQR0);
+  __psq_l1(auStack8,in_GQR0);
+  __psq_l0(auStack24,in_GQR0);
+  __psq_l1(auStack24,in_GQR0);
+  return;
+  #endif
```
Comment (mattbruv) [inline src/Core/p2/iMath3.cpp:110]: I get that this  probably is close to matching, but it's pretty useless as far as readable code goes. If you compare this function to `iSphereIsectVec`, it is much more legible and correct than this. This looks like you basically copied and pasted ghidra's output into a comment, which doesn't really help much honestly. If it almost matches and you want to keep it, this function needs to be cleaned up A LOT. Use the struct definitions to set the properties of a type, not random pointer math everywhere. Basically make this look like `iSphereIsectVec`, and use our custom types everywhere, and format the code with clang-format before I can accept this. Alternatively, you can submit the two other functions in a separate PR and I can merge them (after other issues are addressed), but I don't want to merge this code in

File: src/Core/p2/iMath3.h [hunk 5]
```diff
@@ -1,6 +1,42 @@
...
+	xVec3 dir;
+	float32 min_t;
+	float32 max_t;
+	int flags;
+};
 
+
+#define float32 float
+#define uint32 unsigned int
+
```
Comment (mattbruv) [inline src/Core/p2/iMath3.h]: You should include `<types.h>` instead of re-defining our types

File: src/Core/x/xVec3.cpp [hunk 6]
```diff
@@ -9,7 +9,12 @@
...
 // func_8004E4DC
-#pragma GLOBAL_ASM("asm/Core/x/xVec3.s", "xVec3Copy__FP5xVec3PC5xVec3")
-
+//#pragma GLOBAL_ASM("asm/Core/x/xVec3.s", "xVec3Copy__FP5xVec3PC5xVec3")
+void xVec3Copy(xVec3 * o, const xVec3 * a){
+    float tmp1[] = {((float*)(void*)a)[0],((float*)(void*)a)[1]};
+    float tmp2[] = {((float*)(void*)a)[2],((float*)(void*)a)[3]};
+    *o = *(xVec3*)tmp1;
+    *(float*)(o+0x8)= tmp2[0];
+}
```
Comment (mattbruv) [inline src/Core/x/xVec3.cpp:18]: This doesn't 🆗 right? I can't merge this code if it's uncommented and doesn't build properly. Also, as discussed in discord, this function was probably hand-written assembly, so I highly doubt that's anything close to what they wrote

**Review by mattbruv — CHANGES_REQUESTED** [review]

File: src/Core/p2/iMath3.cpp (hunk 4 above)
Comment (sonich2401) [reply src/Core/p2/iMath3.cpp:110]: Yea I was on 0 sleep when I made this PR. Tbh i shouldn't have made this PR yet. How about I just delete this PR until I have my bearings straight?

**Comment by sonich2401** [comment]
Closed Until I can get my act together and learn how to make a proper PR lol.

**Pull request closed without merging**